src/connectors/toggleRefinement/__tests__/connectToggleRefinement-test.js

```diff
@@ -634,143 +634,271 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/toggle-refi
-          searchParametersBefore,
-          { uiState }
-        );
-        expect(searchParametersAfter).toMatchSnapshot();
-      });
-
-      test('should update the SP base on the UI state (two values)', () => {
-        const [widget, helper, refine] = getInitializedWidget({
-          on: 'free-shipping',
-          off: 'paid-shipping',
-        });
-        refine({ isRefined: false });
-        const uiState = {};
-        const searchParametersBefore = SearchParameters.make(helper.state);
-        const searchParametersAfter = widget.getWidgetSearchParameters(
-          searchParametersBefore,
-          { uiState }
-        );
-        expect(searchParametersAfter).toMatchSnapshot();
-      });
-
-      test('should update the SP base on the UI state - toggled (two values)', () => {
-        const [widget, helper] = getInitializedWidget({
-          on: 'free-shipping',
-          off: 'paid-shipping',
-        });
-        const uiState = {
+        }
+      );
+
+      expect(actual).toEqual({
+        toggle: {
+          discount: true,
+          freeShipping: true,
+        },
+      });
+    });
+  });
+
+  describe('getWidgetSearchParameters', () => {
+    test('returns the `SearchParameters` with the default value without the previous refinement', () => {
+      const render = jest.fn();
+      const makeWidget = connectToggleRefinement(render);
+      const helper = jsHelper({}, '', {
+        disjunctiveFacets: ['freeShipping'],
+        disjunctiveFacetsRefinements: {
+          freeShipping: ['true'],
+        },
+      });
+      const widget = makeWidget({
+        attribute: 'freeShipping',
+      });
+
+      const actual = widget.getWidgetSearchParameters(helper.state, {
+        uiState: {},
+      });
+
+      expect(actual.disjunctiveFacets).toEqual(['freeShipping']);
+      expect(actual.disjunctiveFacetsRefinements).toEqual({
+        freeShipping: [],
+      });
+    });
+
+    test('returns the `SearchParameters` without overriding the disjunctive facets', () => {
+      const render = jest.fn();
+      const makeWidget = connectToggleRefinement(render);
+      const helper = jsHelper({}, '', {
+        disjunctiveFacets: ['freeShipping', 'onSale'],
+        disjunctiveFacetsRefinements: {
+          freeShipping: ['true'],
+        },
+      });
+      const widget = makeWidget({
+        attribute: 'freeShipping',
+      });
+
+      const actual = widget.getWidgetSearchParameters(helper.state, {
+        uiState: {},
+      });
+
+      expect(actual.disjunctiveFacets).toEqual(['freeShipping', 'onSale']);
+      expect(actual.disjunctiveFacetsRefinements).toEqual({
+        freeShipping: [],
+      });
+    });
+
+    test('returns the `SearchParameters` without overriding the other refinements', () => {
+      const render = jest.fn();
+      const makeWidget = connectToggleRefinement(render);
+      const helper = jsHelper({}, '', {
+        disjunctiveFacets: ['freeShipping'],
```

Comment:
```suggestion
        disjunctiveFacets: ['onSale', 'freeShipping'],
```

It's better to have a valid `SearchParameters`.